hir/src/component/mod.rs

```diff
@@ -0,0 +1,284 @@
+use core::{
+    convert::{AsMut, AsRef},
+    ops::{Deref, DerefMut},
+};
+use intrusive_collections::RBTree;
+use std::collections::BTreeMap;
+
+use super::*;
+
+mod interface;
+
+pub use interface::*;
+
+/// Represents the method by which a component function should be invoked in Miden VM
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Default)]
+pub enum FunctionInvocationMethod {
+    /// A function should be invoked by a `call` Miden VM instruction
+    Call,
+    /// A function should be invoked by a `exec` Miden VM instruction
+    #[default]
+    Exec,
+}
+
+/// A component import
+#[derive(Debug)]
+pub struct ComponentImport {
+    /// The interfact function name that is being imported
+    pub interface_function: InterfaceFunctionIdent,
+    /// The component(lifted) type of the imported function
+    pub function_ty: LiftedFunctionType,
+    /// The method of calling the function
+    pub invoke_method: FunctionInvocationMethod,
+    /// The MAST root hash of the function to be used in codegen
+    pub function_mast_root_hash: MastRootHash,
+}
+
+/// The name of a exported function
+#[derive(Debug, Ord, PartialEq, PartialOrd, Eq, Hash, derive_more::From, derive_more::Into)]
```

Comment:
Do you know if `derive_more::Into` corresponds to `impl From<FunctionExportName> for Symbol { .. }` here, or `impl Into<Symbol> for FunctionExportName { .. }`? Unfortunately they aren't equivalent, and we should always do the former, so that the conversions are truly bi-directional (where possible of course, sometimes they are intentionally one-way).

Reply:
Good question! It's `impl From<FunctionExportName> for (Symbol)`.